packages/toolkit/src/query/core/buildThunks.ts

```diff
@@ -513,6 +513,7 @@ export function calculateProvidedByThunk(
     isFulfilled(action) ? action.payload : undefined,
     isRejectedWithValue(action) ? action.payload : undefined,
     action.meta.arg.originalArgs,
+    action.meta,
```

Comment:
Sorry for chiming in later, but I'm like 80% sure that this is the "wrong meta". 

I'm pretty sure this would need to be `action.meta.baseQueryMeta`